From 08ceff9f844ed15ae9eb7aced04a48e864c4bb59 Mon Sep 17 00:00:00 2001
From: Steven Koo <steven.koo@ni.com>
Date: Thu, 30 Jun 2022 19:07:05 +0000
Subject: [PATCH] ni: zynq: Add tftp port selection

Signed-off-by: Steven Koo <steven.koo@ni.com>

---
 include/configs/ni_neon_rev2.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/configs/ni_neon_rev2.h b/include/configs/ni_neon_rev2.h
index 67d838c042..7548a37d61 100644
--- a/include/configs/ni_neon_rev2.h
+++ b/include/configs/ni_neon_rev2.h
@@ -57,5 +57,6 @@
 			"bootm ${load_addr}#conf@zynq-ni-${mboard}.dtb; fi\0" \
 		DFU_ALT_INFO
 
+#define CONFIG_TFTP_PORT
 
 #endif /* __CONFIG_NI_NEON_REV2_H */
