From 396ea2771ce43c27dfde92c6ae26c9296bb921df Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Mon, 9 Nov 2015 14:19:30 -0800
Subject: [PATCH 3/4] zynq: e3xx: Set ethernet mac address.

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 board/ettus/e3xx/board.c | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/board/ettus/e3xx/board.c b/board/ettus/e3xx/board.c
index c31ae3d..b1d0ea58 100644
--- a/board/ettus/e3xx/board.c
+++ b/board/ettus/e3xx/board.c
@@ -88,6 +88,7 @@ static void identify_products(void)
 	u8 mb_buf[sizeof(struct e3xx_mb_eeprom_map)];
 	u16 mb, db;
 	u8 speedgrade;
+	char mstr[20];
 
 	struct e3xx_db_eeprom_map *db_map =
 		(struct e3xx_db_eeprom_map*) &db_buf[0];
@@ -155,17 +156,27 @@ static void identify_products(void)
 	} else if (speedgrade == 3) {
 		if (db == DB_ID_E310_REVB) {
 			setenv("e3xx_type", "e31x_3");
+			//printf("Found %s\n", e3xx_db_names[DB_ID_E310_REVB]);
 		} else if (db == DB_ID_E310_REVC) {
 			setenv("e3xx_type", "e31x_3");
+			//printf("Found %s\n", e3xx_db_names[DB_ID_E310_REVC]);
 		} else if (db == DB_ID_E300_REVB) {
 			setenv("e3xx_type", "e31x_3");
+			//printf("Found %s\n", e3xx_db_names[DB_ID_E300_REVB]);
 		} else if (db == DB_ID_E330_REVC) {
 			setenv("e3xx_type", "e33x_3");
+			//printf("Found %s\n", e3xx_db_names[DB_ID_E330_REVC]);
 		} else {
 			setenv("e3xx_type", "e31x_3");
 			printf("*** Found unknown daughterboard, please update sd card ***\n");
 		}
 	}
+
+	/* grab mac address */
+	sprintf(mstr, "%0X:%0X:%0X:%0X:%0X:%0X",mb_map->mac_addr[0],
+		mb_map->mac_addr[1], mb_map->mac_addr[2], mb_map->mac_addr[3],
+		mb_map->mac_addr[4], mb_map->mac_addr[5]);
+	setenv("ethaddr", mstr);
 }
 
 
-- 
2.4.3

