From f668cd0751981ea6f093466d43787594297bf1c7 Mon Sep 17 00:00:00 2001
From: Geert Uytterhoeven <geert+renesas@glider.be>
Date: Wed, 26 Aug 2015 19:21:00 +0200
Subject: [PATCH 20/47] kbuild: Enable DT symbols when CONFIG_OF_OVERLAY is
 used

Signed-off-by: Geert Uytterhoeven <geert+renesas@glider.be>
---
 scripts/Makefile.lib | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/scripts/Makefile.lib b/scripts/Makefile.lib
index 94133708889d..52a92754bb26 100644
--- a/scripts/Makefile.lib
+++ b/scripts/Makefile.lib
@@ -315,6 +315,10 @@ cmd_dt_S_dtb=						\
 $(obj)/%.dtb.S: $(obj)/%.dtb FORCE
 	$(call if_changed,dt_S_dtb)
 
+ifeq ($(CONFIG_OF_OVERLAY),y)
+DTC_FLAGS += -@
+endif
+
 quiet_cmd_dtc = DTC     $@
 cmd_dtc = $(HOSTCC) -E $(dtc_cpp_flags) -x assembler-with-cpp -o $(dtc-tmp) $< ; \
 	$(DTC) -O $(patsubst .%,%,$(suffix $@)) -o $@ -b 0 \
-- 
2.29.2

