From 752e5382780584927c2a7118a8b5ea1288ee88c6 Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Mon, 3 Apr 2017 20:57:29 -0700
Subject: [PATCH 22/24] fixup! of: Custom printk format specifier for device
 node

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 lib/vsprintf.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/vsprintf.c b/lib/vsprintf.c
index aedf15e..e6236e4 100644
--- a/lib/vsprintf.c
+++ b/lib/vsprintf.c
@@ -1591,7 +1591,7 @@ char *device_node_string(char *buf, char *end, struct device_node *dn,
 				break;
 			case 'r':	/* node reference count */
 				snprintf(tbuf, sizeof(tbuf), "%u",
-					atomic_read(&dn->kobj.kref.refcount));
+					atomic_read(&dn->kobj.kref.refcount.refs));
 				append_str(tbuf, pass, &len, &buf, end);
 				break;
 			default:
-- 
2.7.4

