From 6a335edc4bdeaa15af9a70a7b783307a5fe85fe0 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@opensdr.com>
Date: Tue, 4 Feb 2014 12:38:07 +0100
Subject: [PATCH 2/3] E300 : Disable QSPI.

Signed-off-by: Philip Balister <philip@opensdr.com>
---
 include/configs/zynq_zc70x.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/configs/zynq_zc70x.h b/include/configs/zynq_zc70x.h
index a550701..a79d060 100644
--- a/include/configs/zynq_zc70x.h
+++ b/include/configs/zynq_zc70x.h
@@ -20,7 +20,6 @@
 
 #define CONFIG_ZYNQ_SDHCI0
 #define CONFIG_ZYNQ_USB
-#define CONFIG_ZYNQ_QSPI
 #define CONFIG_ZYNQ_I2C0
 #define CONFIG_ZYNQ_EEPROM
 #define CONFIG_ZYNQ_BOOT_FREEBSD
-- 
1.7.11.7

