From 14d902e33745899e267a39d4f5f928883d58f571 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Mon, 19 Aug 2013 14:49:33 -0400
Subject: [PATCH] mach-zynq : Add dt entry for e200.

Signed-off-by: Philip Balister <philip@balister.org>
---
 arch/arm/mach-zynq/common.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-zynq/common.c b/arch/arm/mach-zynq/common.c
index 1352940..2acbd7d 100644
--- a/arch/arm/mach-zynq/common.c
+++ b/arch/arm/mach-zynq/common.c
@@ -222,6 +222,7 @@ static const char * const xilinx_dt_match[] = {
 	"xlnx,zynq-zc702",
 	"xlnx,zynq-zc706",
 	"xlnx,zynq-zc770",
+	"xlnx,ettus-e200",
 	NULL
 };
 
-- 
1.8.3.1

