From 2fa75b3b4ddb3c0114d0a512ac3a555597b0d3e0 Mon Sep 17 00:00:00 2001
From: Joerg Hofrichter <joerg.hofrichter@emerson.com>
Date: Tue, 4 Nov 2025 15:36:18 +0100
Subject: [PATCH] try to import ssl_match_hostname from urllib3.util

---
 salt/ext/tornado/netutil.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/salt/ext/tornado/netutil.py b/salt/ext/tornado/netutil.py
index f86b430..bf0ef33 100644
--- a/salt/ext/tornado/netutil.py
+++ b/salt/ext/tornado/netutil.py
@@ -54,9 +54,14 @@ if hasattr(ssl, 'match_hostname') and hasattr(ssl, 'CertificateError'):  # pytho
 elif ssl is None:
     ssl_match_hostname = SSLCertificateError = None  # type: ignore
 else:
-    import backports.ssl_match_hostname
-    ssl_match_hostname = backports.ssl_match_hostname.match_hostname
-    SSLCertificateError = backports.ssl_match_hostname.CertificateError  # type: ignore
+    try:
+        import urllib3.util.ssl_match_hostname
+        ssl_match_hostname = urllib3.util.ssl_match_hostname.match_hostname
+        SSLCertificateError = urllib3.util.ssl_match_hostname.CertificateError  # type: ignore
+    except ModuleNotFoundError:
+        import backports.ssl_match_hostname
+        ssl_match_hostname = backports.ssl_match_hostname.match_hostname
+        SSLCertificateError = backports.ssl_match_hostname.CertificateError  # type: ignore
 
 if hasattr(ssl, 'SSLContext'):
     if hasattr(ssl, 'create_default_context'):
