From c99989254a88b19c8d6664b35ca650ae19bc0913 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Wed, 11 Sep 2013 11:14:07 -0400
Subject: [PATCH] board.c : unlock/lock the slcr around the code that resets
 the gpios/

Signed-off-by: Philip Balister <philip@balister.org>
---
 board/xilinx/zynq/board.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/board/xilinx/zynq/board.c b/board/xilinx/zynq/board.c
index 5f6988b..e1da270 100644
--- a/board/xilinx/zynq/board.c
+++ b/board/xilinx/zynq/board.c
@@ -62,6 +62,8 @@ static void reset_usb_phy()
 {
 	unsigned int mask, tmp;
 
+	zynq_slcr_unlock();
+
 	/* Set USB_MIO to output and enable it */
 	tmp = readl(MIO_BASE + DIRM_0);
 	printf("Orig DIRM: %X\n", tmp);
@@ -86,6 +88,8 @@ static void reset_usb_phy()
 	udelay(1);
 	writel(mask | USB_MIO | ETH_MIO, MIO_BASE + MASK_DATA_0_LSW);
 	udelay(2);
+
+	zynq_slcr_lock();
 }
 
 static void ethaddr_init(void)
-- 
1.8.3.1

