From 82ce4bc1cbd183e2432d374b4edbefdc93f08f6d Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Mon, 24 Jun 2013 14:20:20 -0400
Subject: [PATCH 4/4] Also reset the Ethernet phy.

Signed-off-by: Philip Balister <philip@balister.org>
---
 board/xilinx/zynq/board.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/board/xilinx/zynq/board.c b/board/xilinx/zynq/board.c
index 5e1dfe6..5f6988b 100644
--- a/board/xilinx/zynq/board.c
+++ b/board/xilinx/zynq/board.c
@@ -56,6 +56,7 @@ Xilinx_desc fpga100 = XILINX_XC7Z100_DESC(0x100);
 #define OEN_0		0x208
 
 #define USB_MIO 9
+#define ETH_MIO 11
 
 static void reset_usb_phy()
 {
@@ -65,23 +66,25 @@ static void reset_usb_phy()
 	tmp = readl(MIO_BASE + DIRM_0);
 	printf("Orig DIRM: %X\n", tmp);
 	tmp |= (1<<USB_MIO);
+	tmp |= (1<<ETH_MIO);
 	printf("Updated DIRM: %X\n", tmp);
 	writel(tmp, MIO_BASE + DIRM_0);
 
 	tmp = readl(MIO_BASE + OEN_0);
 	printf("Orig OEN: %X\n", tmp);
 	tmp |= (1<<USB_MIO);
+	tmp |= (1<<ETH_MIO);
 	printf("Updated OEN: %X\n", tmp);
 	writel(tmp, MIO_BASE + OEN_0);
 
 	/* Pulse USB_MIO low and leave it hi */
-	mask = (~(1<<USB_MIO)<<16);
+	mask = (~((1<<USB_MIO) | (1<<ETH_MIO))<<16);
 	printf("Mask set to: %X\n", mask);	
-	writel(mask | USB_MIO, MIO_BASE + MASK_DATA_0_LSW);
+	writel(mask | USB_MIO | ETH_MIO, MIO_BASE + MASK_DATA_0_LSW);
 	udelay(1);
 	writel(mask, MIO_BASE + MASK_DATA_0_LSW);
 	udelay(1);
-	writel(mask | USB_MIO, MIO_BASE + MASK_DATA_0_LSW);
+	writel(mask | USB_MIO | ETH_MIO, MIO_BASE + MASK_DATA_0_LSW);
 	udelay(2);
 }
 
-- 
1.8.3.1

