From 87d6268052bf6b9172fbd6fe00942bb087ac88f2 Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Mon, 19 Aug 2013 14:49:33 -0400
Subject: [PATCH] mach-zynq : Add dt entry for e200.

Signed-off-by: Philip Balister <philip@balister.org>
---
 arch/arm/mach-zynq/common.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/arch/arm/mach-zynq/common.c b/arch/arm/mach-zynq/common.c
index 1710fcc..b6c4051 100644
--- a/arch/arm/mach-zynq/common.c
+++ b/arch/arm/mach-zynq/common.c
@@ -217,6 +217,7 @@ static void zynq_system_reset(char mode, const char *cmd)
 static const char * const zynq_dt_match[] = {
 	"xlnx,zynq-zc770",
 	"xlnx,zynq-7000",
+	"xlnx,ettus-e200",
 	NULL
 };
 
-- 
1.8.3.1

