From cd3d558d39e0408cf7f1ab96b2dac6aa14333f0e Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?J=C3=B6rg=20Hofrichter?= <joerg.hofrichter@emerson.com>
Date: Thu, 23 Oct 2025 09:40:49 +0200
Subject: [PATCH] layer: Add scarthgap to compatible layers
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Signed-off-by: Jörg Hofrichter <joerg.hofrichter@emerson.com>
---
 conf/layer.conf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/conf/layer.conf b/conf/layer.conf
index c40376a..7ba32f3 100644
--- a/conf/layer.conf
+++ b/conf/layer.conf
@@ -15,4 +15,4 @@ BBFILE_PRIORITY_openamp-layer = "5"
 # cause compatibility issues with other layers
 LAYERVERSION_openamp-layer = "1"
 
-LAYERSERIES_COMPAT_openamp-layer = "rocko sumo thud warrior zeus kirkstone"
+LAYERSERIES_COMPAT_openamp-layer = "rocko sumo thud warrior zeus kirkstone scarthgap"
-- 
2.34.1

