From f59e79f055dff238f1371a6d62b66e8509a3e159 Mon Sep 17 00:00:00 2001
From: Bjoern Bachmann <bjoern.bachmann@emerson.com>
Date: Wed, 25 Mar 2026 17:09:59 +0100
Subject: [PATCH] Add mender to environment variables.

---
 include/configs/ni-x410.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/include/configs/ni-x410.h b/include/configs/ni-x410.h
index 7e521521c5e..b8b2e08e7a2 100644
--- a/include/configs/ni-x410.h
+++ b/include/configs/ni-x410.h
@@ -26,14 +26,16 @@
         "load_addr=0x2000000\0" \
         "bootargs=uio_pdrv_genirq.of_id=usrp-uio at24.io_limit=64  rootwait clk_ignore_unused\0" \
         "emmcboot=" \
+          "run mender_setup; " \
           "setenv disk_name eMMC && " \
-          "setenv uboot_root \"mmc 0:2\" && " \
-          "setenv kernel_root \"/dev/mmcblk0p2\" && " \
+          "setenv uboot_root ${mender_uboot_root} && " \
+          "setenv kernel_root ${mender_kernel_root} && " \
           "run disk_bootcmd\0" \
         "sdboot=" \
+          "run mender_setup; " \
           "setenv disk_name SD && " \
-          "setenv uboot_root \"mmc 1:2\" && " \
-          "setenv kernel_root \"/dev/mmcblk1p2\" && " \
+          "setenv uboot_root ${mender_uboot_root} && " \
+          "setenv kernel_root ${mender_kernel_root} && " \
           "run disk_bootcmd\0" \
         "disk_bootcmd=run swsync_disk; " \
           "echo Copying FIT from ${disk_name} to RAM... && " \
-- 
2.43.0

