From afddcc9f6686b85380f2c4ac15f901012b725b1e Mon Sep 17 00:00:00 2001
From: Joerg Hofrichter <joerg.hofrichter@ni.com>
Date: Wed, 26 Jun 2019 08:04:50 +0000
Subject: [PATCH] resolve: also add missing.h (needed for musl)

Signed-off-by: Joerg Hofrichter <joerg.hofrichter@ni.com>
---
 src/resolve/resolvectl.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/resolve/resolvectl.c b/src/resolve/resolvectl.c
index 1d2fc71309..b12c80f125 100644
--- a/src/resolve/resolvectl.c
+++ b/src/resolve/resolvectl.c
@@ -30,6 +30,7 @@
 #include "strv.h"
 #include "terminal-util.h"
 #include "verbs.h"
+#include "missing.h"
 
 static int arg_family = AF_UNSPEC;
 static int arg_ifindex = 0;
-- 
2.17.1

