From 3eabf0db94b33a3054218c672522cadb92dbeee6 Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Fri, 3 Nov 2017 17:07:21 -0700
Subject: [PATCH 14/15] fixup! board: ni: zynq: Add parsing code for sulfur

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 board/ni/zynq/sulfur-eeprom.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/board/ni/zynq/sulfur-eeprom.c b/board/ni/zynq/sulfur-eeprom.c
index 713d5b8ccfe..d2f1916594f 100644
--- a/board/ni/zynq/sulfur-eeprom.c
+++ b/board/ni/zynq/sulfur-eeprom.c
@@ -18,6 +18,7 @@ struct sulfur_xboard_match {
 
 static const struct sulfur_xboard_match db_match[] = {
 	{ .pid = 0x150, .name = "Magnesium XCVR" },
+	{ .pid = 0x152, .name = "Rhodium XCVR" },
 	{ .pid = 0x180, .name = "EISCAT RX" },
 	{},
 };
@@ -76,10 +77,6 @@ int sulfur_probe_dboard(int dboard)
 
 	crc = crc32(0, (const u8*)&eeprom, sizeof(eeprom) - 4);
 	if (htonl(crc) != eeprom.crc) {
-		/*
-		printf("%s: CRC for eeprom doesn't match! %08x vs %08x\n",
-		       __func__, htonl(crc), eeprom.crc);
-		       */
 		printf("Slot %u: Unitialized\n", dboard);
 		return -EINVAL;
 	}
@@ -125,7 +122,7 @@ int sulfur_probe_mboard(void)
 			sulfur_get_mboard_name(htons(eeprom.pid)),
 			'A' + htons(eeprom.rev),
 			eeprom.serial);
-	return 0;
+	return htons(eeprom.rev);
 }
 
 int sulfur_read_ethaddr(unsigned char *ethaddr, int which)
-- 
2.14.2

