From 99626fc779bca6a41c89a68431b4e9bf7569f24d Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Fri, 31 Mar 2017 15:38:07 -0700
Subject: [PATCH 21/45] misc: eeprom: Hardcode the io_length constraint to 16

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 drivers/misc/eeprom/at24.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/misc/eeprom/at24.c b/drivers/misc/eeprom/at24.c
index 051b14766ef9..3d94f69ea493 100644
--- a/drivers/misc/eeprom/at24.c
+++ b/drivers/misc/eeprom/at24.c
@@ -91,7 +91,7 @@ struct at24_data {
  *
  * This value is forced to be a power of two so that writes align on pages.
  */
-static unsigned io_limit = 128;
+static unsigned io_limit = 16;
 module_param(io_limit, uint, 0);
 MODULE_PARM_DESC(io_limit, "Maximum bytes per I/O (default 128)");
 
-- 
2.14.2

